packages/@sanity/form-builder/src/inputs/files/ImageInput/ImagePreview.tsx

```diff
@@ -12,13 +18,45 @@ interface Props {
   alt: string
 }
 
+/* 
+  Used for setting the initial image height - specifically for images 
+  that are small and so can take less space in the document
+*/
+const getImageSize = (src) => {
+  const imageUrlParams = new URLSearchParams(src.split('?')[1])
+  const rect = imageUrlParams.get('rect')
+
+  if (rect) {
+    return [rect.split(',')[2], rect.split(',')[3]].map(Number)
+  }
+
+  return src.split('-')[1].split('.')[0].split('x').map(Number)
+}
+
 export function ImagePreview(props: ComponentProps<typeof Card> & Props) {
   const {drag, readOnly, isRejected, src, ...rest} = props
   const [isLoaded, setLoaded] = useState(false)
-
+  const [rootElement, setRootElement] = useState<HTMLDivElement | null>(null)
+  const rootRect = useElementRect(rootElement)
+  const rootWidth = rootRect?.width || 0
   const acceptTone = isRejected || readOnly ? 'critical' : 'primary'
   const tone = drag ? acceptTone : 'default'
 
+  const maxHeightToPx = (MAX_DEFAULT_HEIGHT * document.documentElement.clientHeight) / 100 // convert from vh to px, max height of the input
```

Comment:
Could we wrap `maxHeightToPx` in a useMemo here? It never change after initial render if I'm reading this correctly.